_blog/2024-02-09-using-clerk-in-rust.mdx

```diff
@@ -0,0 +1,1014 @@
+    .await;
+
+    if create_query.is_err() {
+        return HttpResponse::InternalServerError().json(serde_json::json!({
+            "status":"FAILED",
+            "message":"Failed to create an issue"
+        }));
+    }
+
+    HttpResponse::Ok().json(serde_json::json!({
+        "status":"SUCCESS",
+        "message":"Created the issue successfully"
+    }))
+}
+
+```
+
+### Retrieve all issues endpoint
+
+This endpoint will retrieve all the issues from the database:
+
+```rust
+#[get("/issues")]
+async fn get_issues(state: web::Data<AppState>) -> impl Responder {
+    let query: Result<Vec<Issue>, sqlx::Error> = sqlx::query_as("SELECT * FROM issues")
+        .fetch_all(&state.pool)
+        .await;
+
+    let issues = match query {
+        Ok(value) => value,
+        Err(e) => {
+            return HttpResponse::InternalServerError().json(serde_json::json!({
+                "status": "FAILED",
+                "message": e.to_string(),
+            }));
+        }
+    };
+
+    HttpResponse::Ok().json(issues)
+}
+
+```
+
+### Get issue by `id` endpoint
+
+This endpoint function will extract the `issue_id` slug from the path using the **[Path extractor](https://actix.rs/docs/extractors)** from Actix web and query our database where the issue `id` is equal to the `issue_id` slug of the path.
+
+```rust
+#[get("/issue/{issue_id}")]
+async fn get_issue(state: web::Data<AppState>, path: web::Path<i32>) -> impl Responder {
+    let issue_id = path.into_inner();
+
+    let query: Result<Issue, sqlx::Error> = sqlx::query_as("SELECT * FROM issues WHERE id=$1")
+        .bind(issue_id)
+        .fetch_one(&state.pool)
+        .await;
+
+    let issue = match query {
+        Ok(value) => value,
+        Err(_) => {
+            return HttpResponse::InternalServerError().json(serde_json::json!({
+                "status":"FAILED",
+                "message":"Something went wrong."
+            }));
+        }
+    };
+
+    HttpResponse::Ok().json(serde_json::json!({
+        "status": "SUCCESS",
+        "data": issue,
+    }))
+}
+
+```
+
+### Update issue by `id` endpoint
+
+To update an issue by id we will use slug from the path which will contain the `issue_id` to query our database and update the record if it exists in the database. Before updating the record we can query if the record exists in the database or not:
+
+```rust
+#[patch("/issue/{issue_id}")]
+async fn update_issue(
+    payload: web::Json<NewIssue>,
+    state: web::Data<AppState>,
+    path: web::Path<i32>,
+    req: HttpRequest,
+) -> impl Responder {
+    let issue_id = path.into_inner();
+
+    let srv_req = ServiceRequest::from_request(req);
+
+    let claim = match clerk_authorize(&srv_req, &state.client, true).await {
+        Ok(value) => value.1,
+        Err(_) => {
+            return HttpResponse::Unauthorized().json(serde_json::json!({
+                "status":"Failed",
+                "message":"Unauthorized"
+            }));
+        }
+    };
```

Comment:
It could be worth extracting this logic to a function since it is repeated several times in the endpoints.